File: app/src/main/java/com/example/myapplication/api/models/ApiResponse.java
package com.example.myapplication.api.models;

public class ApiResponse<T> {
    private boolean success;
    private T data;
    private Error error;

    public boolean isSuccess() { return success; }
    public T getData() { return data; }
    public Error getError() { return error; }

    public static class Error {
        private String code;
        private String message;

        public String getCode() { return code; }
        public String getMessage() { return message; }
    }
}

File: ClinicAPI/src/API/DTOs/ApiResponse.cs
namespace API.DTOs;

public class ApiResponse<T>
{
    public bool Success { get; set; }
    public T? Data { get; set; }
    public ErrorDetail? Error { get; set; }

    public static ApiResponse<T> SuccessResponse(T data)
    {
        return new ApiResponse<T>
        {
            Success = true,
            Data = data
        };
    }

    public static ApiResponse<T> ErrorResponse(string code, string message)
    {
        return new ApiResponse<T>
        {
            Success = false,
            Error = new ErrorDetail
            {
                Code = code,
                Message = message
            }
        };
    }
}

public class ErrorDetail
{
    public string Code { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
}
